src/i18n/index.ts

```diff
@@ -0,0 +1,7 @@
+import enLocale from "./locales/en.json";
+
+type Messages = Record<string, Record<string, string>>;
```

Comment:
This type assumes depth of nesting is one. Maybe in a future iteration, it can have more than one level of depth. Not a blocker, just pointing it out so we can enhance it in future if need be.